python/lsst/ts/m2/mock_server.py

```diff
@@ -0,0 +1,428 @@
+
+    async def _process_message_command(self):
+        """Process the incoming message of command."""
+
+        try:
+            msg_input = await asyncio.wait_for(
+                self.server_command.reader.readuntil(separator=tcpip.TERMINATOR),
+                self.TIMEOUT_IN_SECOND,
+            )
+
+            msg = self._decode_and_deserialize_json_message(msg_input)
+
+            # Process the command
+            name = msg["id"]
+            if self._is_command(name):
+
+                # Acknowledge the command
+                sequence_id = msg["sequence_id"]
+                await self._acknowledge_command(sequence_id)
+
+                await self._process_command(msg)
+
+                # Command result (only reply the success at this moment)
+                await self._reply_command(sequence_id)
+
+            # TODO (DM-30851):
+            # Handle the event at DM-30851
+
+        except asyncio.TimeoutError:
+            await asyncio.sleep(self.TIMEOUT_IN_SECOND)
+
+    def _decode_and_deserialize_json_message(self, msg_encode):
+        """Decode the incoming JSON message and return a dictionary.
+
+        Parameters
+        ----------
+        msg_encode : `bytes` or `None`
+            Encoded message.
+
+        Returns
+        -------
+        `dict`
+            Decoded messge. If the msg_encode is None or can not be decoded by
+            JSON, return an empty dictionary.
+        """
+
+        try:
+            return json.loads(msg_encode.decode()) if msg_encode is not None else dict()
+
+        except json.JSONDecodeError:
+            self.log.debug(f"Can not decode the message: {msg_encode}.")
+            return dict()
+
+    def _is_command(self, message_name):
+        """Is the command or not.
+
+        Parameters
+        ----------
+        message_name : `str`
+            Message name in the header.
+
+        Returns
+        -------
+        `bool`
+            True if this is a command.
+        """
+
+        return message_name.startswith("cmd_")
+
+    async def _acknowledge_command(self, sequence_id):
+        """Acknowledge the command with the sequence ID.
+
+        Parameters
+        ----------
+        sequence_id : `int`
+            Sequence ID that should be >= 0.
+        """
+
+        id_ack = CommandStatus.Ack
+        msg_ack = {"id": id_ack.name.lower(), "sequence_id": sequence_id}
+        await write_json_packet(self.server_command.writer, msg_ack)
+
+    async def _process_command(self, message):
+        """Process the command.
+
+        Parameters
+        ----------
+        message : `dict`
+            Command message.
+        """
+
+        command_name = message["id"]
+        if command_name in self._command_response:
+            await self._command_response[command_name](message, self._message_event)
+        else:
+            commands = list(self._command_response.keys())
+            self.log.debug(
```

Comment:
Do you think logging a debug message is enough here? Consider raising a `RuntimeError` exception.

Reply:
This is to be consistent with the behavior of M2 server, which will ignore the unrecognized command instead of raising an error. I thought it may not be good to raise an error to stop the control system. Let us keep this behavior at this moment and I could change it in a latter time if it is needed. Thanks!